src/cast_core/src/CrudActor.cpp

```diff
@@ -1117,19 +1117,24 @@ struct CommitTransactionOperation : public BaseOperation {
 
 struct WithTransactionOperation : public BaseOperation {
     WithTransactionOperation(const Node& opNode,
-                             bool onSession,
+                             bool,
                              CollectionHandle collectionHandle,
                              metrics::Operation operation,
                              PhaseContext& context,
                              ActorId id)
         : BaseOperation(context, opNode),
-          _onSession{onSession},
           _collectionHandle{std::move(collectionHandle)},
           _operation{operation} {
+        auto& onSession = opNode["OnSession"];
+        if (!onSession || !onSession.to<bool>()){
+            BOOST_THROW_EXCEPTION(InvalidConfigurationException(
+                "'withTransaction' requires explicit 'OnSession: true'."));
+        };
+        _onSession = onSession.to<bool>();
         auto& opsInTxn = opNode["OperationsInTransaction"];
         if (!opsInTxn.isSequence()) {
             BOOST_THROW_EXCEPTION(InvalidConfigurationException(
-                "'WithTransaction' requires an 'OperationsInTransaction' node of sequence type."));
+                "'withTransaction' requires an 'OperationsInTransaction' node of sequence type."));
```

Comment:
This error message is difficult to understand!

Reply:
Yeah I agree, but I don't know how to fix the overall message. I was just fixing the typo here.

Reply:
"OperationsInTransaction should be a list of operations" or "withTransaction expected a OperationsInTransaction with a list of operations" if "withTransaction" or a LOC isn't otherwise in the error message. Do either of those match with your understanding of the code here?

"OperationsInTransaction" probably is always a list in all contexts it's in (guessing it's just this one), and when you're reading the YAML it makes more sense as describing it as containing a list rather than it being a node of sequence type (which makes more sense if you're assembling the AST)

Reply:
Yup I agree with you and have updated the error message. Thanks!